### ISSUE TITLE
Display error messsage when API fails

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-toolpad#1570]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I was trying to use the Stripe API expansion feature https://stripe.com/docs/api/expanding_objects so move from 101 request to 1 for getting the order ID on the stripe disputes. As it turns out, I didn't use the REST API correctly, Stripe was returning:

```json
{
  "error": {
    "message": "This property cannot be expanded (payment_intent). You may want to try expanding 'data.payment_intent' instead.",
    "request_log_url": "https://dashboard.stripe.com/logs/req_kAgNt46d3WYxxp?t=1673969095",
    "type": "invalid_request_error"
  }
}
```

Stripe are effectively telling me how to solve the problem. However, Toolpad hides the information: https://master--toolpad.mui.com/_toolpad/app/cld06azhx0001k59tpypy3aoy/pages/ib03nog

<img width="548" alt="Screenshot 2023-01-17 at 16 47 04" src="https://user-images.githubusercontent.com/3165635/212944476-f8be8441-eae8-4fca-a531-3fa1b7209ba9.png">

**Comment by Janpot** [comment]
I agree, we need show the body of the response somewhere, even if it returned with a bad HTTP status code. That would be helpful in debugging.
